server/src/services/AnnouncementServices.js

```diff
@@ -32,27 +22,45 @@ const AnnouncementServices = {
         if (err) {
           reject(err);
         } else {
+          let removeIndex;
           if (
-            listOfCompleted.every((value) => {
-              return value._id != announcement._id;
+            listOfCompleted.every((value, index) => {
+              if (value.id === announcement.id) {
+                removeIndex = index;
+              }
+              return value.id != announcement.id;
             })
           ) {
-            listOfCompleted.push(announcement._id);
+            listOfCompleted.push(announcement.id);
+            resolve(
+              UserModel.findOneAndUpdate(
+                { _id: currentUser._id },
+                { completedAnnouncements: listOfCompleted },
+                (err, user) => {
+                  if (err || !user) {
+                    reject('UNABLE_TO_UPDATE_USER');
+                  } else {
+                    resolve(user);
+                  }
+                },
+              ),
+            );
+          } else {
+            listOfCompleted.splice(removeIndex, 1);
+            resolve(
+              UserModel.findOneAndUpdate(
+                { _id: currentUser._id },
+                { completedAnnouncements: listOfCompleted },
+                (err, user) => {
+                  if (err || !user) {
+                    reject('UNABLE_TO_UPDATE_USER');
+                  } else {
+                    resolve(user);
+                  }
+                },
+              ),
+            );
```

Comment:
This should be refactored so that only the `listOfCompleted.push` and `listOfCompleted.splice` operations are contained in the if-else block. The rest of the logic seems to be the same between the two, so it should be separated out.

Also, you should only ever call `resolve()` once in a Promise, as that is when the function will return - here your Promise is resolving to a new function which is then calling the old `resolve()` callback when it finishes. So, instead of `resolve(UserModel. ...)`, you only need to call `resolve()` inside the callback on the `UserModel.findOneAndUpdate()` method